crates/plex-api/src/media_container/server/library.rs

```diff
@@ -0,0 +1,496 @@
+
+    pub grandparent_title: Option<String>,
+    pub grandparent_studio: Option<String>,
+    pub grandparent_year: Option<u32>,
+    pub grandparent_content_rating: Option<String>,
+    pub grandparent_index: Option<u32>,
+
+    pub grandparent_thumb: Option<String>,
+    pub grandparent_art: Option<String>,
+    pub grandparent_theme: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum MetadataType {
+    Movie,
+    Episode,
+    Photo,
+    Show,
+    Artist,
+    Album,
+    Collection,
+    Season,
+    Track,
+    Playlist,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "tests_deny_unknown_fields", serde(deny_unknown_fields))]
+#[serde(rename_all = "camelCase")]
+pub struct Metadata {
+    pub key: String,
+    #[serde(deserialize_with = "deserialize_number_from_string")]
+    pub rating_key: u32,
+    pub guid: String,
+
+    #[serde(rename = "type")]
+    pub metadata_type: Option<MetadataType>,
+    pub subtype: Option<String>,
+
+    pub title: String,
+    pub title_sort: Option<String>,
+    pub original_title: Option<String>,
+    pub studio: Option<String>,
+    pub year: Option<u32>,
+    pub content_rating: Option<String>,
+    pub summary: Option<String>,
+    pub rating: Option<f32>,
+    pub rating_image: Option<String>,
+    pub audience_rating: Option<f32>,
+    pub audience_rating_image: Option<String>,
+    pub tagline: Option<String>,
+    pub duration: Option<u64>,
+    pub originally_available_at: Option<Date>,
+
+    pub thumb: Option<String>,
+    pub art: Option<String>,
+    pub theme: Option<String>,
+    pub composite: Option<String>,
+    pub banner: Option<String>,
+
+    pub index: Option<u32>,
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
+    pub child_count: Option<u32>,
+    pub leaf_count: Option<u32>,
+    pub viewed_leaf_count: Option<u32>,
+    pub skip_children: Option<bool>,
+
+    pub view_count: Option<u64>,
+    pub skip_count: Option<u64>,
+    #[serde(default, with = "time::serde::timestamp::option")]
+    pub last_viewed_at: Option<OffsetDateTime>,
+
+    #[serde(rename = "createdAtTZOffset")]
+    pub created_at_tz_offset: Option<String>,
+    pub created_at_accuracy: Option<String>,
+    #[serde(with = "time::serde::timestamp")]
+    pub added_at: OffsetDateTime,
+    #[serde(default, with = "time::serde::timestamp::option")]
+    pub deleted_at: Option<OffsetDateTime>,
+    #[serde(default, with = "time::serde::timestamp::option")]
+    pub updated_at: Option<OffsetDateTime>,
+    #[serde(default, deserialize_with = "deserialize_option_number_from_string")]
+    pub loudness_analysis_version: Option<u32>,
+    #[serde(default, deserialize_with = "optional_boolish")]
+    pub has_premium_primary_extra: Option<bool>,
+    pub view_offset: Option<u64>,
+    pub chapter_source: Option<ChapterSource>,
+    pub primary_extra_key: Option<String>,
+
+    #[serde(flatten)]
```

Comment:
Wow! Great catch! I wouldn't have thought about using `flatten` for that! :D

Reply:
I had wanted to use the same struct for both and experimented with `serde_with::with_prefix` which worked well except you can't use `deny_unknown_fields` with it unfortunately.